lib/common/client.ts

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect } from "react";
+import type { TUseStoryblokState } from "../types";
+import { registerStoryblokBridge } from "@storyblok/js";
+
+export const useStoryblokState: TUseStoryblokState = (
+  initialStory = null,
+  bridgeOptions = {}
+) => {
+  let [story, setStory] = useState(initialStory);
+
+  const isBridgeEnable =
+    typeof window !== "undefined" &&
+    typeof window.storyblokRegisterEvent !== "undefined";
+
+  if (!isBridgeEnable || !initialStory) {
+    return initialStory;
+  }
+
+  useEffect(() => {
+    setStory(initialStory);
+    registerStoryblokBridge(
```

Comment:
Do you really need to use `useEffect` ? I think `registerStoryblokBridge` should be enough, if I'm not wrong the function is subscribing to `window.storyblokRegisterEvent` and you pass the callback to set your newStory.

Reply:
And if the initialStory(props) change, the component should re-render.